### PR TITLE
Feat event attendees api

### LINKED ISSUE #72: Fixing /events/{eventId}/users
Get all the users concerned with the given event.

Response Object 
{}

### PR DESCRIPTION
**Pull request by Krace33**
Fixes #72 
#70 also included

### CODE REVIEW

File: src/main/java/com/pecacm/backend/services/SupportService.java [hunk 1]
```diff
@@ -59,13 +59,11 @@ public Pair<List<EventUserDetails>, List<EventUserDetails>> getNonParticipants(I
...
     }
 
-    public Page<EventUserDetails> getEventParticipants(Event event, PageRequest pageRequest) {
-        List<EventUserDetails> participants = transactionRepository.findByEventIdAndRole(event.getId(), EventRole.PARTICIPANT, pageRequest)
+    public List<EventUserDetails> getEventParticipants(Integer eventId, PageRequest pageRequest) {
+        return transactionRepository.findByEventIdAndRole(eventId, EventRole.PARTICIPANT, pageRequest)
                 .stream().map(transaction -> {
                     User user = transaction.getUser();
                     return new EventUserDetails(user.getId(), user.getEmail(), user.getName(), user.getDp());
-                }).collect(Collectors.toList());
```
Comment (harasees-singh) [inline src/main/java/com/pecacm/backend/services/SupportService.java:67]: nit: can reduce to .toList()

File: src/main/java/com/pecacm/backend/services/EventService.java [hunk 2]
```diff
@@ -60,20 +63,24 @@ public List<Event> getEventsByBranch(Branch branch) {
...
-        Optional<User> user = userRepository.findByEmail(email);
-        List<Event> events = new ArrayList<>();
-        if (user.isEmpty()) {
-            throw new AcmException(ErrorConstants.USER_NOT_FOUND, HttpStatus.NOT_FOUND);
-        }
+    public EventAttendeesResponse getEventUsersByRole(Integer eventId, EventRole eventRole, PageRequest pageRequest) {
         if (eventRole == null) {
-            transactionRepository.findByUserId(user.get().getId(), PageRequest.of(offset, pageSize))
-                    .forEach(transaction -> events.add(transaction.getEvent()));
+            Pair<List<EventUserDetails>, List<EventUserDetails>> nonParticipants = supportService.getNonParticipants(eventId);
```
Comment (harasees-singh) [inline src/main/java/com/pecacm/backend/services/EventService.java:68]: nit: please change var name, nonParticipants is not reader friendly

Comment (Krace33) [reply src/main/java/com/pecacm/backend/services/EventService.java]: Changed to organizingTeam

File: src/main/java/com/pecacm/backend/services/EventService.java [hunk 3]
```diff
@@ -60,20 +63,24 @@ public List<Event> getEventsByBranch(Branch branch) {
...
-        Optional<User> user = userRepository.findByEmail(email);
-        List<Event> events = new ArrayList<>();
-        if (user.isEmpty()) {
-            throw new AcmException(ErrorConstants.USER_NOT_FOUND, HttpStatus.NOT_FOUND);
-        }
+    public EventAttendeesResponse getEventUsersByRole(Integer eventId, EventRole eventRole, PageRequest pageRequest) {
         if (eventRole == null) {
-            transactionRepository.findByUserId(user.get().getId(), PageRequest.of(offset, pageSize))
-                    .forEach(transaction -> events.add(transaction.getEvent()));
+            Pair<List<EventUserDetails>, List<EventUserDetails>> organizingTeam = supportService.getNonParticipants(eventId);
```
Comment (harasees-singh) [inline src/main/java/com/pecacm/backend/services/EventService.java:68]: please be consistent with naming, organizingTeam has been used synonymously with nonParticipants

**Review by harasees-singh — APPROVED** [review]

**Pull request merged**